### PR TITLE
Some potential ways to track FROM image digests

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

**Review by jerith — COMMENTED** [review]
Is this for tracking which image tags points to which digests so we can detect changes?

Is this for tracking which images are/were/will be built from which upstream digests?

Are these separate problems that potentially need different solutions?

**Comment by JayH5** [comment]
@jerith currently I am thinking of this as just updating the Dockerfile (and possibly associated metadata) in a repo. Whether that results in an updated image pushed to a registry or not is kind of a separate problem (but I'm not sure... maybe the two should be done together?). I'm not sure if that answers your question?

So the answer I think is _"for tracking which image tags points to which digests so we can detect changes"_.

File: specifying-digests.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+* Docker will definitely build your image using exactly the same `FROM` image every time.
+* Native Docker, not defining some new thing.
+
+### Disadvantages:
+* Tag information no longer checked. Could become incorrect.
+* Adding the digest to tags is not commonly done.
+* A bit clunky.
+
+## Digest in a "metadata" comment
+It could be possible just to add a comment to the Dockerfile with the image digest. This wouldn't be picked up by Docker itself, but a separate tool could. Most importantly, the Git repository would track the digest of the `FROM` tag.
```
Comment (jerith) [inline specifying-digests.md]: Minor phrasing issue: something like "the digest is tracked in the git repo" would make it clearer that the repo itself isn't doing any magic to track this, the maintainer is putting the information in the repo.

**Review by jerith — APPROVED** [review]
:+1:

**Pull request merged**